server/src/main/java/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsService.kt

```diff
@@ -1,9 +1,18 @@
 package org.jetbrains.bsp.bazel.server.diagnostics
 
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+import ch.epfl.scala.bsp4j.Diagnostic
 import ch.epfl.scala.bsp4j.PublishDiagnosticsParams
+import ch.epfl.scala.bsp4j.TextDocumentIdentifier
 import java.nio.file.Path
+import java.util.concurrent.ConcurrentHashMap
 
-class DiagnosticsService(workspaceRoot: Path) {
+/**
+ * @property hasAnyProblems keeps track of problems in given file so BSP reporter
+ * can publish diagnostics with an empty array, to clear up former diagnostics.
+ * see: https://youtrack.jetbrains.com/issue/BAZEL-376
+ */
+class DiagnosticsService(workspaceRoot: Path, val hasAnyProblems: MutableMap<String, Set<TextDocumentIdentifier>>) {
```

Comment:
why `val hasAnyProblems` is public?

Reply:
Oh, I forgot to make it private and expose the `problems` via getter method: done https://github.com/JetBrains/bazel-bsp/pull/381/commits/cd029b9e12ba05c21e36e6f911c32789b97d841a Thank you for pointing out